### PR TITLE
tests of js/api/filters.js (continued), handlers.js and parser.js

### PR DESCRIPTION
**Pull request by zezutom**
Please note the questions in TODO comments

### CODE REVIEW

File: test/js/api/filters-spec.js [hunk 1]
```diff
@@ -185,4 +185,38 @@ describe('api/filters', function() {
...
         assert.equal(expected, filterCss(['css.all']));
     });
+
+    it('should escape special characters', function() {
+        const actual = 'stra$nge?name*for.a.fi*le',
+            expected = 'stra\\$nge\\?name\\*for\\.a\\.fi\\*le';
+        assert.equal(expected, filters.apply(actual, ['escape']));
+    });
+
+    //it('TODO: regexp - how is it used?', function() {});
```
Comment (aalbuquerque73) [inline test/js/api/filters-spec.js]: We've discuss this, just adding it here for visibility

regexp filter reverts the escape filter on the text between two '&rx;'. This allows us to use regular expressions with xml files that contain characters that need to be escaped

**Comment by aalbuquerque73** [comment]
## Note:

 api/handlers is used when saving the request/response to disk
api/parser is used when reading from disk

all the supported mime types use the same api:
- `handler.has` tests if the handler module supports the mime type
- `handler.apply` executes (applies) the respective handler of mime type to the data passed
- `parser.has` tests if the parser module supports the mime type
- `parser.apply` executes (applies) the respective parser of mime type to the data passed

**Comment by zezutom** [comment]
@aalbuquerque73 - Thanks for your comments. Please review the updated tests.

PS: I believe there is no need for explicit tests to handler.apply and parser.apply, as these two functions are used extensively throughout the specs.

**Comment by zezutom** [comment]
Closing this off, since there have been extensive changes to the base since this PR was created.

**Pull request closed without merging**